### PR TITLE
Vet clinic database: add "join table" for visits

### PR DESCRIPTION
**Pull request by Alfred-KInyua**
### The objectives required 

- Understand the different types of relationships between tables.
- Prepare complex queries that answer analytical questions.
- Use primary key & foreign key mechanism for joining tables.

#### I was able to answer the following questions 

- Who was the last animal seen by William Tatcher?
- How many different animals did Stephanie Mendez see?
- List all vets and their specialties, including vets with no specialties.
- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
- What animal has the most visits to vets?
- Who was Maisy Smith's first visit?
- Details for most recent visit: animal information, vet information, and date of visit.
- How many visits were with a vet that did not specialize in that animal's species?
- What specialty should Maisy Smith consider getting? Look for the species she gets the most

v1

v2

v3
v4
v5

v6
v7
v8

v9

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -16,3 +16,16 @@ create table animals(
...
+	age int,
+	date_of_graduation date);
+    CREATE TABLE specializations (
+	vet_id INTEGER REFERENCES vets(id),
+	species_id INTEGER REFERENCES species(id)
+);
+
+create table visits (vet_id int references vets(id),animal_id int references animals(id),date_of_visit date;
+);
+create table specializations (vet_id int references vets(id),species_id int references species(id));
```
Comment (mateo951) [inline schema.sql:31]: All new tables are added with the propper columns and data types ✔

File: data.sql [hunk 2]
```diff
@@ -55,4 +55,30 @@ select * from animals where escape_attempts =0;
 select avg(weight_in_kg) from animals;
 select name min(weight_in_kg) and max(weight_in_kg) from animals group by name;
 select neutered, max(escape_attempts) from animals group by neutered;
-select name, avg(escape_attempts) from animals where date_of_birth BETWEEN '1990-1-1' AND '2000-12-31' group by name;
\ No newline at end of file
+select name, avg(escape_attempts) from animals where date_of_birth BETWEEN '1990-1-1' AND '2000-12-31' group by name;
+
```
Comment (mateo951) [inline data.sql:59]: Please be sure to add the following data to your vets table as it is part of the requirements 🤓

image

**Review by mateo951 — CHANGES_REQUESTED** [review]
### Hi @Alfred-KInyua, 

**🔥Highlights🔥**
- Tables creation ✔
- Queries ✔

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

File: data.sql (hunk 2 above)
Comment (Alfred-KInyua) [reply data.sql:59]: @mateo951 Thanks for the review, I populated my DB  with vets data but forgot to type out my queries in vscode

**Review by SajjadAhmad14 — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

### Hi Kinyua, 
## Congratulations 🎉 Thank you for implementing all the requirements 🙏
### Please go ahead and merge the branch!

## Happy Coding 👏👏👏
#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**